src/panels/lovelace/editor/hui-entity-row-editor.ts

```diff
@@ -0,0 +1,343 @@
+    });
+  }
+
+  public toggleMode() {
+    this.GUImode = !this.GUImode;
+  }
+
+  public refreshYamlEditor(focus = false) {
+    if (this._configElement?.refreshYamlEditor) {
+      this._configElement.refreshYamlEditor(focus);
+    }
+    if (!this._yamlEditor?.codemirror) {
+      return;
+    }
+    this._yamlEditor.codemirror.refresh();
+    if (focus) {
+      this._yamlEditor.codemirror.focus();
+    }
+  }
+
+  protected render(): TemplateResult {
+    return html`
+      <div class="wrapper">
+        ${this.GUImode
+          ? html`
+              <div class="gui-editor">
+                ${this._loading
+                  ? html`
+                      <ha-circular-progress
+                        active
+                        alt="Loading"
+                        class="center margin-bot"
+                      ></ha-circular-progress>
+                    `
+                  : this._configElement}
+              </div>
+            `
+          : html`
+              <div class="yaml-editor">
+                <ha-code-editor
+                  mode="yaml"
+                  autofocus
+                  .value=${this.yaml}
+                  .error=${Boolean(this._error)}
+                  .rtl=${computeRTL(this.hass)}
+                  @value-changed=${this._handleYAMLChanged}
+                  @keydown=${this._ignoreKeydown}
+                ></ha-code-editor>
+              </div>
+            `}
+        ${this._error
+          ? html`
+              <div class="error">
+                ${this._error}
+              </div>
+            `
+          : ""}
+        ${this._warnings
+          ? html`
+              <div class="warning">
+                UI editor is not supported for this config:
+                <br />
+                <ul>
+                  ${this._warnings.map((warning) => html`<li>${warning}</li>`)}
+                </ul>
+                You can still edit your config in yaml.
+              </div>
+            `
+          : ""}
+      </div>
+    `;
+  }
+
+  protected updated(changedProperties) {
+    super.updated(changedProperties);
+    if (this._configElement && changedProperties.has("hass")) {
+      this._configElement.hass = this.hass;
+    }
+  }
+
+  private _handleUIConfigChanged(ev: UIRowConfigChangedEvent) {
+    ev.stopPropagation();
+    const config = ev.detail.config;
+    this.value = config;
+  }
+
+  private _handleYAMLChanged(ev) {
+    ev.stopPropagation();
+    const newYaml = ev.detail.value;
+    if (newYaml !== this.yaml) {
+      this.yaml = newYaml;
+    }
+  }
+
+  private async _updateConfigElement(): Promise<void> {
+    if (!this.value) {
+      return;
+    }
+
+    let rowType: string;
+    if (!this.value.type && "entity" in this.value) {
+      // @ts-ignore
+      const domain = this.value.entity.split(".", 1)[0];
+      rowType = `${
+        DOMAIN_TO_ELEMENT_TYPE![domain] ||
+        DOMAIN_TO_ELEMENT_TYPE!._domain_not_found
+      }-entity`;
```

Comment:
If one of these types. I think I should set it to just be one a type of generic or something. BEcause it will see the type change and try and reload the same element again

Reply:
Uuuh whut?

Reply:
There is no type on the entity row for regular entities. The entity file comes from the Domain to Element Type. Well all of those domains will have the same editor file that needs loaded

Reply:
Oooh... can we make the check on the editor element instead of on the type?

Reply:
What do you mean? We get the editor element from the type?

Reply:
Yeah, and all those types return the same element right?

Reply:
They should. at least right now. My thought was to always load this editor. And if it exists in that then we don't need to get the config element from the element its self.

Reply:
I don't like that it is not future proof, but also can't think of a case that will need it, so we'll it when it happens I guess